src/db.c

```diff
@@ -1777,12 +1904,14 @@ int bzmpopGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult
  * follow in SQL-alike style. Here we parse just the minimum in order to
  * correctly identify keys in the "STORE" option. */
 int sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *result) {
-    int i, j, num, *keys, found_store = 0;
+    int i, j, num, found_store = 0;
+    keyReference *keys;
     UNUSED(cmd);
 
     num = 0;
     keys = getKeysPrepareResult(result, 2); /* Alloc 2 places for the worst case. */
-    keys[num++] = 1; /* <sort-key> is always present. */
+    keys[num].pos = 1; /* <sort-key> is always present. */
+    keys[num++].flags = CMD_KEY_READ;
```

Comment:
This seems like an independent bug that maybe we can pull out of this PR.

Reply:
if this wasn't about a security topic, i would mind leaving it behind... but since it is, i think we must handle it.
we can leave it out of this PR, and handle in a followup PR after this one is merged.
in which case, please add a ToDo checkbox in the top comment, to open a separate PR / issue for it.